src/resolve-config.js

```diff
@@ -19,23 +21,43 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function resolveConfig(filePath, opts, sync) {
```

Comment:
If you'd like to merge these two functions internally, I'd recommend to rename this to something like `_resolveConfig`, and it'll be called by `resolveConfig`/`resolveConfig.sync` so that we won't expose an unnecessary parameter that may cause problems in the future.

Reply:
Oh, good idea! That also prevents us from exposing an API that is *sometimes* synchronous, depending on the arguments... http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony

Fixed in https://github.com/prettier/prettier/pull/2760/commits/f5f382b661d93310843d7f459d2a962405a1ff89